core/src/main/scala/sttp/tapir/TapirAuth.scala

```diff
@@ -34,8 +35,14 @@ object TapirAuth {
       challenge: WWWAuthenticateChallenge
   ): EndpointInput.Auth[T, EndpointInput.AuthType.Http] = {
     val codec = implicitly[Codec[List[String], T, CodecFormat.TextPlain]]
-    val authCodec =
-      Codec.list(Codec.string.map(stringPrefixWithSpace(authScheme))).mapDecode(codec.decode)(codec.encode).schema(codec.schema)
+    val authCodec = Codec
+      .listFiltered[String, String, TextPlain](
```

Comment:
Won't sth like `Codec.id.map(filter)(identity).map(stringPrefixWithSpace(authScheme)).mapDecode(...).schema(...)` be simpler?